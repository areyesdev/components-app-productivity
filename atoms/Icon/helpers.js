const ICONS_ASSETS_PATH = '/icons',
  iconMap = {
    arrowRight: 'arrow-right',
    angleDown: 'angle-down',
    iconPlus: 'icon-plus',
  },
  iconSize = {
    sm: 15,
    md: 26,
    lg: 40,
    xl: 40,
  }

export const mapType = (type) => `${ICONS_ASSETS_PATH}/${iconMap[type]}.svg`
export const mapSize = (size) => iconSize[size]
