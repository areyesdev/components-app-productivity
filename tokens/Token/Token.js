import React from 'react'

import { Color, Spacing, FontFamily, FontSize, FontWeight } from './components'

const Token = () => <></>

Token.Color = Color
Token.Spacing = Spacing
Token.FontFamily = FontFamily
Token.FontSize = FontSize
Token.FontWeight = FontWeight

export default Token
