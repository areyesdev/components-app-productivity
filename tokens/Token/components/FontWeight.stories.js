import Token from '..'
import { choices } from '../..'

import { getTemplate, getListTemplate } from '../../../helpers/storybook'
import { getTokenItems } from '../helpers'

const Template = getTemplate(Token.FontWeight)
const ListTemplate = getListTemplate(Token.FontWeight)

export default {
  title: 'Tokens/FontWeight',
  component: Token.FontWeight,
  args: {
    children: 'Cheap doors deny rune Mithrandir spawning brown dishes?',
  },
  argTypes: {
    value: { control: null },
    label: { control: null },
  },
}

const getItems = getTokenItems(choices.fontWeight, 'fontWeight')

export const Default = Template.bind({})
Default.args = {
  label: 'var(--font-weight-normal)',
  value: choices.fontWeight.normal,
}

export const Sizes = ListTemplate.bind({})
Sizes.args = { items: getItems() }
