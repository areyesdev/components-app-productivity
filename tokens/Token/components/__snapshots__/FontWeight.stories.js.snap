// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Tokens/FontWeight Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="container"
  >
    <div
      className="font-weight"
      style={
        Object {
          "fontWeight": 400,
        }
      }
    >
      Cheap doors deny rune Mithrandir spawning brown dishes?
    </div>
    <div
      className="description"
    >
      <span
        className="label"
      >
        var(--font-weight-normal)
      </span>
      <span
        className="label"
      >
        400
      </span>
    </div>
  </div>
</div>
`;

exports[`Storyshots Tokens/FontWeight Sizes 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="container"
  >
    <div
      className="font-weight"
      style={
        Object {
          "fontWeight": 100,
        }
      }
    >
      Cheap doors deny rune Mithrandir spawning brown dishes?
    </div>
    <div
      className="description"
    >
      <span
        className="label"
      >
        var(--font-weight-hairline)
      </span>
      <span
        className="label"
      >
        100
      </span>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="font-weight"
      style={
        Object {
          "fontWeight": 200,
        }
      }
    >
      Cheap doors deny rune Mithrandir spawning brown dishes?
    </div>
    <div
      className="description"
    >
      <span
        className="label"
      >
        var(--font-weight-thin)
      </span>
      <span
        className="label"
      >
        200
      </span>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="font-weight"
      style={
        Object {
          "fontWeight": 300,
        }
      }
    >
      Cheap doors deny rune Mithrandir spawning brown dishes?
    </div>
    <div
      className="description"
    >
      <span
        className="label"
      >
        var(--font-weight-light)
      </span>
      <span
        className="label"
      >
        300
      </span>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="font-weight"
      style={
        Object {
          "fontWeight": 400,
        }
      }
    >
      Cheap doors deny rune Mithrandir spawning brown dishes?
    </div>
    <div
      className="description"
    >
      <span
        className="label"
      >
        var(--font-weight-normal)
      </span>
      <span
        className="label"
      >
        400
      </span>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="font-weight"
      style={
        Object {
          "fontWeight": 500,
        }
      }
    >
      Cheap doors deny rune Mithrandir spawning brown dishes?
    </div>
    <div
      className="description"
    >
      <span
        className="label"
      >
        var(--font-weight-medium)
      </span>
      <span
        className="label"
      >
        500
      </span>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="font-weight"
      style={
        Object {
          "fontWeight": 600,
        }
      }
    >
      Cheap doors deny rune Mithrandir spawning brown dishes?
    </div>
    <div
      className="description"
    >
      <span
        className="label"
      >
        var(--font-weight-semibold)
      </span>
      <span
        className="label"
      >
        600
      </span>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="font-weight"
      style={
        Object {
          "fontWeight": 700,
        }
      }
    >
      Cheap doors deny rune Mithrandir spawning brown dishes?
    </div>
    <div
      className="description"
    >
      <span
        className="label"
      >
        var(--font-weight-bold)
      </span>
      <span
        className="label"
      >
        700
      </span>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="font-weight"
      style={
        Object {
          "fontWeight": 800,
        }
      }
    >
      Cheap doors deny rune Mithrandir spawning brown dishes?
    </div>
    <div
      className="description"
    >
      <span
        className="label"
      >
        var(--font-weight-extrabold)
      </span>
      <span
        className="label"
      >
        800
      </span>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="font-weight"
      style={
        Object {
          "fontWeight": 900,
        }
      }
    >
      Cheap doors deny rune Mithrandir spawning brown dishes?
    </div>
    <div
      className="description"
    >
      <span
        className="label"
      >
        var(--font-weight-black)
      </span>
      <span
        className="label"
      >
        900
      </span>
    </div>
  </div>
</div>
`;
